### DISCUSSION TITLE
Import issues & comment timestamps during migration

### DISCUSSION THREAD

**Discussion by inopinatus** [community/community#24839] [Apps, API and Webhooks]
<p>I’m spiking a migration of our private repos to Github from Bitbucket (source &amp; bugs) and Trello (kanban boards). Also spiking Gitlab, we’ll end up choosing depending on how much my team like the outcomes.</p>
<p>Obviously source is easy to move, no dramas there.</p>
<p>For accuracy of the historical record I need to ensure all the issues and comments have the original timestamps, and appear to be written by the original author (all of whom are also in the new Github org).</p>
<p>Seems like an obvious need, but I don’t see anything in Github APIs v3 or v4 for creating issues with specified author and correct creation time.</p>
<p>I reckon I can overcome the authorship problem by collecting API keys from my users, but the timestamp problem eludes me. Can it be done? Or will we have to append the timestamps as crude plain text to each comment &amp; issue?</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>Thanks for reaching out and we’re glad that you’re considering GitHub.</p>
<p>There isn’t a way to override the timestamps on issues via the API. Events will appear to have happened when the API request was submitted. You may want to add a standard block to the issues or comments using <a href="https://help.github.com/en/articles/about-writing-and-formatting-on-github">Markdown</a> that gives the original timestamp for reference. Maybe something like:</p>
<pre><code>&gt; **Originally submitted at:** 2019-01-01 12:31:19
</code></pre>
<p>which would render like this:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/c/cad469c6f104a99e97b6fae978c1cbc3a120d92d.png" data-download-href="/uploads/short-url/sWjxCcliY8j7s4Y9brRpKMvFLw9.png?dl=1" title="Screen Shot 2019-09-25 at 3.56.08 PM.png"><img src="https://user-images.githubusercontent.com/1038121/181087044-568b4410-66dc-4e9b-b127-b0af66d48975.png" alt="Screen Shot 2019-09-25 at 3.56.08 PM.png" data-base62-sha1="sWjxCcliY8j7s4Y9brRpKMvFLw9" width="690" height="313" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/c/cad469c6f104a99e97b6fae978c1cbc3a120d92d_2_690x313.png, https://aws1.discourse-cdn.com/github/original/2X/c/cad469c6f104a99e97b6fae978c1cbc3a120d92d.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/c/cad469c6f104a99e97b6fae978c1cbc3a120d92d.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/c/cad469c6f104a99e97b6fae978c1cbc3a120d92d_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2019-09-25 at 3.56.08 PM.png</span><span class="informations">1020×464 20 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>So hopefully less crude than you feared :grinning:</p>
<p>Let us know if you have more questions.</p>

**Comment by inopinatus** [comment]
<p>Hi <a class="mention" href="/lee-dohm">@lee-dohm</a> and thanks for your followup. The resolution in this case was all the boards stayed on Trello, and bugs will start afresh. Timeline and history matter, and losing that as structured data wasn’t acceptable to the team.</p>
